### PR TITLE
Support cipher suite limitation for REST API

### PR DESCRIPTION
**Pull request by bluthg**
Many environments require a limitation of allowed TLS cipher suites / levels.
See e.g. the german BSI requirements: 
https://www.bsi.bund.de/SharedDocs/Downloads/EN/BSI/Publications/TechGuidelines/TG02102/BSI-TR-02102-2.pdf?__blob=publicationFile&v=10

This implements an optional "ciphers" setting that - if given - enforces the ciphers on the REST API socket.

See also #1730.

### CODE REVIEW

File: patroni/config.py [hunk 1]
```diff
@@ -265,7 +265,7 @@ def _set_section_values(section, params):
                 if value:
                     ret[section][param] = value
 
-        _set_section_values('restapi', ['listen', 'connect_address', 'certfile', 'keyfile', 'cafile', 'verify_client',
+        _set_section_values('restapi', ['listen', 'connect_address', 'certfile', 'keyfile', 'cafile', 'ciphers', 'verify_client',
```
Comment (CyberDem0n) [inline patroni/config.py:268]: Flake8 complains: `patroni/config.py:268:121: E501 line too long (129 > 120 characters)`.

You need to move ` 'verify_client',` to the next line.

File: docs/SETTINGS.rst [hunk 2]
```diff
@@ -322,6 +322,7 @@ REST API
         -  **certfile**: (optional): Specifies the file with the certificate in the PEM format. If the certfile is not specified or is left empty, the API server will work without SSL.
         -  **keyfile**: (optional): Specifies the file with the secret key in the PEM format.
         -  **cafile**: (optional): Specifies the file with the CA_BUNDLE with certificates of trusted CAs to use while verifying client certs.
+        -  **ciphers**: (optional): Specifies the permitted cipher suites (e.g. "ECDHE-RSA-AES256-GCM-SHA384:DHE-RSA-AES256-GCM-SHA384:ECDHE-RSA-AES128-GCM-SHA256:DHE-RSA-AES128-GCM-SHA256:!SSLv1:!SSLv2:!SSLv3:!TLSv1:!TLSv1.1")
```
Comment (CyberDem0n) [inline docs/SETTINGS.rst:325]: Please also update the [ENVIRONMENT.rst](https://github.com/zalando/patroni/blob/master/docs/ENVIRONMENT.rst#rest-api).

**Review by CyberDem0n — CHANGES_REQUESTED** [review]

**Comment by Jan-M** [comment]
:+1:

**Comment by CyberDem0n** [comment]
:+1:

**Review by CyberDem0n — APPROVED** [review]

**Pull request merged**